scipy/special/_basic.py

```diff
@@ -2421,16 +2422,22 @@ def factorial2(n, exact=False):
     array(105.00000000000001)
     >>> factorial2(7, exact=True)
     105
+    >>> factorial2(-3, exact=True)
+    -1
 
     """
     if exact:
-        if n < -1:
-            return 0
-        if n <= 0:
-            return 1
-        val = 1
-        for k in range(n, 0, -2):
-            val *= k
+        if not isscalar(n):
```

Comment:
use camel case variable name

Reply:
This is not a variable. See https://numpy.org/doc/stable/reference/generated/numpy.isscalar.html